wait/health_test.go

```diff
@@ -0,0 +1,99 @@
+package wait
+
+import (
+	"context"
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/go-connections/nat"
+	tcexec "github.com/testcontainers/testcontainers-go/exec"
+)
+
+type healthStrategyTarget struct {
+	Health *types.Health
+}
+
+func (st healthStrategyTarget) Host(ctx context.Context) (string, error) {
+	return "", nil
+}
+
+func (st healthStrategyTarget) Ports(ctx context.Context) (nat.PortMap, error) {
+	return nil, nil
+}
+
+func (st healthStrategyTarget) MappedPort(ctx context.Context, n nat.Port) (nat.Port, error) {
+	return n, nil
+}
+
+func (st healthStrategyTarget) Logs(ctx context.Context) (io.ReadCloser, error) {
+	return nil, nil
+}
+
+func (st healthStrategyTarget) Exec(ctx context.Context, cmd []string, options ...tcexec.ProcessOption) (int, io.Reader, error) {
+	return 0, nil, nil
+}
+
+func (st healthStrategyTarget) State(ctx context.Context) (*types.ContainerState, error) {
+	return &types.ContainerState{Health: st.Health}, nil
+}
+
+// TestWaitForHealthTimesOutForUnhealthy confirms that an unhealthy container will eventually
+// time out.
+func TestWaitForHealthTimesOutForUnhealthy(t *testing.T) {
+	target := healthStrategyTarget{Health: &types.Health{Status: types.Unhealthy}}
+	wg := NewHealthStrategy().WithStartupTimeout(100 * time.Millisecond)
+	err := wg.WaitUntilReady(context.Background(), target)
+
+	if err != nil && err == context.DeadlineExceeded {
+		return
+	}
+	t.Fatal(err)
```

Comment:
Let's rewrite this block as below:
```suggestion
	assert.NotNil(t, err)
	assert.True(t, errors.Is(err, context.DeadlineExceeded))
```